plugins/woocommerce-blocks/assets/js/blocks/checkout/password-strength-meter/index.tsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import { sprintf, __ } from '@wordpress/i18n';
+import { useInstanceId } from '@wordpress/compose';
+import { passwordStrength } from 'check-password-strength';
+import { usePrevious } from '@woocommerce/base-hooks';
+import { useEffect } from '@wordpress/element';
+import clsx from 'clsx';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+declare global {
+	interface Window {
+		zxcvbn: ( password: string ) => {
+			score: number;
+		};
+	}
+}
+
+const scoreDescriptions = [
+	__( 'Too weak', 'woocommerce' ),
+	__( 'Weak', 'woocommerce' ),
+	__( 'Medium', 'woocommerce' ),
+	__( 'Strong', 'woocommerce' ),
+	__( 'Very strong', 'woocommerce' ),
+];
+
+/**
+ * Renders a password strength meter.
+ *
+ * Uses zxcvbn to calculate the password strength if available, otherwise falls back to check-password-strength which
+ * does not include dictionaries of common passwords.
+ */
+const PasswordStrengthMeter = ( {
+	password = '',
+	onChange,
+}: {
+	password: string;
+	onChange?: ( strength: number ) => void;
+} ): JSX.Element | null => {
```

Comment:
`JSX.Element` is deprecated and the preferred way is `React.ReactElement` though I don't think this should block this PR, feel free to change it if you like.